stream-chat-android-client/src/main/java/io/getstream/chat/android/client/uploader/StreamFileUploader.kt

```diff
@@ -38,106 +47,115 @@ internal class StreamFileUploader(
+        channelType = channelType,
+        channelId = channelId,
+        file = file.asBodyPart(),
+        progressCallback = null,
+    ).execute().map(UploadFileResponse::toUploadedFile)
 
     override fun deleteFile(
         channelType: String,
         channelId: String,
         userId: String,
         url: String,
-    ): Result<Unit> {
-        return retrofitCdnApi.deleteFile(
-            channelType = channelType,
-            channelId = channelId,
-            url = url,
-        ).execute().toUnitResult()
-    }
+    ): Result<Unit> = retrofitCdnApi.deleteFile(
+        channelType = channelType,
+        channelId = channelId,
+        url = url,
+    ).execute().toUnitResult()
 
     override fun deleteImage(
         channelType: String,
         channelId: String,
         userId: String,
         url: String,
-    ): Result<Unit> {
-        return retrofitCdnApi.deleteImage(
-            channelType = channelType,
-            channelId = channelId,
-            url = url,
-        ).execute().toUnitResult()
+    ): Result<Unit> = retrofitCdnApi.deleteImage(
+        channelType = channelType,
+        channelId = channelId,
+        url = url,
+    ).execute().toUnitResult()
+
+    override fun uploadFile(
+        file: File,
+        user: User?,
+        progressCallback: ProgressCallback?,
+    ): Result<UploadedFile> = retrofitCdnApi.uploadFile(
+        file = file.asBodyPart(),
+        user = user?.asBodyPart(),
+        progressCallback = progressCallback,
+    ).execute().map(UploadFileResponse::toUploadedFile)
+
+    override fun deleteFile(
+        url: String,
+        userId: UserId?,
+    ): Result<Unit> = retrofitCdnApi.deleteFile(url = url)
+        .execute().toUnitResult()
+
+    override fun uploadImage(
+        file: File,
+        user: User?,
+        progressCallback: ProgressCallback?,
+    ): Result<UploadedFile> = retrofitCdnApi.uploadImage(
+        file = file.asBodyPart(),
+        user = user?.asBodyPart(),
+        progressCallback = progressCallback,
+    ).execute().map(UploadFileResponse::toUploadedFile)
+
+    override fun deleteImage(
+        url: String,
+        userId: UserId?,
+    ): Result<Unit> = retrofitCdnApi.deleteImage(url = url)
+        .execute().toUnitResult()
+
+    private fun File.asBodyPart(): MultipartBody.Part {
+        val body = asRequestBody(contentType = getMediaType())
+        val filename = filenameSanitizer.sanitize(filename = name)
+        return MultipartBody.Part.createFormData(name = "file", filename = filename, body = body)
+    }
+
+    private fun User.asBodyPart(): MultipartBody.Part = with(dtoMapping) {
+        val adapter = Moshi.Builder().build().adapter(UpstreamUserDto::class.java)
+        val json = adapter.toJson(toDto())
+        val body = json.toRequestBody(contentType = "application/json".toMediaType())
+        MultipartBody.Part.createFormData(name = "user", filename = null, body = body)
     }
```

Comment:
Do we need to map the full user entity here? :thinking:

Reply:
Just the upstream DTO

https://getstream.github.io/protocol/#/product%3Achat/UploadFileGlobal

https://github.com/GetStream/stream-chat-js/blob/5a1bd6d2a1d8d832bbd83f93f7936d4fe074a529/src/types.ts#L887

Reply:
I think it is only a "easy to use" implementation, but I don't think backend need the full `UpstreamUserDto`
<img width="916" height="674" alt="Screenshot from 2025-09-08 11-15-02" src="https://github.com/user-attachments/assets/6907f517-e0e0-4239-8882-0ff8f2f02885" />

I assume they would only need the userId, and maybe they are using json notation, but something like the following would be enough. 
```
user: { id: 123 }
```

Reply:
The user is now entirely removed as it is not needed to be sent from the mobile clients